parley/src/layout/alignment.rs

```diff
@@ -89,12 +97,21 @@ pub(crate) fn align<B: Brush>(
                 }
             }
         }
+
+        if is_rtl {
+            // In RTL text, trailing whitespace is on the left. As we hang that whitespace, offset
+            // the line to the left.
+            line.metrics.offset -= line.metrics.trailing_whitespace;
+        }
     }
 }
 
 /// Removes previous justification applied to clusters.
 /// This is part of resetting state in preparation for re-linebreaking the same layout.
 pub(crate) fn unjustify<B: Brush>(layout: &mut LayoutData<B>) {
```

Comment:
I think `unjustify` is broken (and was broken prior to this, as it calculates `extra` / `free_space` differently). I intend to propose a simplification in a follow-up PR.